sqltypes/json.go

```diff
@@ -0,0 +1,35 @@
+package sqltypes
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+
+	"github.com/upfluence/errors"
+)
+
+type JSONValue[T any] struct {
```

Comment:
is there a point having a generic here? 

what about having just `Data` being simply a `interface{}` as i mentioned in my comment?

The upside is of keeping it like this is that it assign a default value for Data
The downside is that we have to drop the support for all go version < 1.18

Reply:
The problem is for the `Scan` the unmarshall need a defined type, otherwise it default to `map[string]interface{}`. That way we can json encode/decode any object. I though it would be useful in a lot more cases.

For the 1.18, it's a bit sad I agree but at least it will force the update to 1.18 ? :sweat_smile: